tools/swift-inspect/Sources/swift-inspect/DarwinRemoteProcess.swift

```diff
@@ -179,20 +198,27 @@ internal final class DarwinRemoteProcess: RemoteProcess {
                                                                Self.ReadBytes,
                                                                Self.GetStringLength,
                                                                Self.GetSymbolAddress) else {
+      runCleanups()
       return nil
     }
     self.context = context
+    cleanups.append({ swift_reflection_destroyReflectionContext(context) })
 
     _ = CSSymbolicatorForeachSymbolOwnerAtTime(self.symbolicator, kCSNow, { owner in
       let address = CSSymbolOwnerGetBaseAddress(owner)
       _ = swift_reflection_addImage(self.context, address)
     })
   }
```

Comment:
I wonder if we can instead do a local variable and defer the execution. On the success path we just push that to the instance wide cleanup and empty out the local cleanups. This would allow us to avoid having to invoke `runCleanups()` before returning `nil` which is still error prone.

Reply:
I thought about this a little more and decided that the way to work with the language rather than against it is to wrap these things in a helper that ensures they get cleaned up, and drop this `deinit` entirely. Then we don't have to worry about whether `deinit` is called or not on any particular path, and we don't need explicit cleanup calls anywhere.

I've added a noncopyable `Cleanup` struct that takes care of this, and rearranged this code to use it.